pkg/build/api/v1beta3/types.go

```diff
@@ -382,6 +386,21 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty" description:"forces the source build to pull the image if true"`
 }
 
+// BuildPostCommitSpec holds a build post commit hook specification.
+type BuildPostCommitSpec struct {
+	// Shell is an action that executes a shell script in a container
+	// running the build output image.
+	Shell string `json:"shell,omitempty" description:"shell script to be executed in a container running the build output image"`
```

Comment:
I played around with how `/bin/sh -c` handles extra arguments:

```console
$ /bin/sh -c 'echo "$@"; echo $0; echo $1; echo $2'

/bin/sh



$ /bin/sh -c 'echo "$@"; echo $0; echo $1; echo $2' foo bar
bar
foo
bar

$ /bin/sh -c 'echo "$@"; echo $0; echo $1; echo $2' -- foo bar
foo bar
--
foo
bar
$ /bin/sh -c 'echo "$@"; echo $0; echo $1; echo $2' /bin/sh foo bar
foo bar
/bin/sh
foo
bar
```

As `man bash` says, the extra arguments are treated as positional arguments starting with `$0`.

So in case the hook is specified  as:

```go
hook := BuildPostCommitSpec{
	Script: "echo $0\necho $1\necho $2\nmvn test \"$@\"",
	Args: []string{"foo", "bar"},
}
```

We'd start a container with:

```go
containerOpts := docker.CreateContainerOptions{
	Config: &docker.Config{
		Entrypoint: {"/bin/sh", "-c"},
		Args: {"echo $0\necho $1\necho $2\nmvn test \"$@\"", "/bin/sh", "foo", "bar"},
		//                                                   ^^^^^^^^^
	},
}
```

As per the manual, since we're using `-c`, it doesn't matter what comes after the script, it will be treated as positional arguments. Instead of `"--"`, using `"/bin/sh"` will set `$0` to what it would be if there were no arguments.

Just a note, this still requires some minimal shell knowledge, e.g. setting the script to `mvn test $@` is not the same as `mvn test "$@"`.